drivers/pcie/host/controller.c

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2021 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(pcie_core, LOG_LEVEL_INF);
+
+#include <kernel.h>
+#include <drivers/pcie/pcie.h>
+#include <drivers/pcie/controller.h>
+
+#if CONFIG_PCIE_MSI
+#include <drivers/pcie/msi.h>
+#endif
+
+/* arch agnostic PCIe API implementation */
+
+uint32_t pcie_conf_read(pcie_bdf_t bdf, unsigned int reg)
+{
+	const struct device *dev;
+
+	dev = DEVICE_DT_GET(DT_ALIAS(pcie0));
```

Comment:
This looks like it should be a chosen node. Aliases in Zephyr are generally reserved for samples (think `led0`). Why is this an alias?

Reply:
@mbolivar-nordic ok, I just followed what Linux does, busses can be specified as aliases, `chosen` is exclusively use by bootloaders

The DT spec goes in this side

Reply:
@mbolivar-nordic so according to Zephyr doc, I should a new `zephyr,pcie-controller` Zephyr-specific chosen property ?

Reply:
OK done

Reply:
> @mbolivar-nordic so according to Zephyr doc, I should a new `zephyr,pcie-controller` Zephyr-specific chosen property ?

That's what we generally do when we want to pick a node for use by a subsystem, yeah. Thanks.